pkgs/applications/networking/cluster/kfctl/default.nix

```diff
@@ -28,5 +30,6 @@ buildGoModule rec {
     homepage = "https://github.com/kubeflow/kfctl";
     license = licenses.asl20;
     maintainers = with maintainers; [ mvnetbiz ];
+    broken = stdenv.isDarwin; # build fails with go > 1.17
```

Comment:
I think `badPlatforms` is the preferred way to mark a package broken like this.

Reply:
`badPlatforms` is ambiguous and undocumented.

Reply:
Not sure what you mean by ambiguous, but it's the most common way to mark a package broken conditionally.

Reply:
> but it's the most common way to mark a package broken conditionally.

Doesn't seem to be.

Reply:
If you really want to argue about this...
```
$ ack -l badPlatform | wc -l
73
git log -G badPlatforms --oneline | wc -l
167
$ ack -l 'broken = !?is\w+' | wc -l
5
```

`badPlatforms` is better than assertions or making a package broken because it can be queried with [lib.availableOn](https://github.com/NixOS/nixpkgs/blob/a5983f7bff6981c94ff90b514a26bf2018a90c60/lib/meta.nix#L91-L99). This is used for enabling/disabling build features based on the availability of dependency.

Reply:
```
$ ack -l 'broken = !?is\w+' | wc -l
5
```
Maybe try looking at this one again?

Reply:
Fine, but it's still worse than badPlatforms. You could have implemented the suggestion in half the time we spent arguing...

Reply:
Orthogonal to this PR, I would be interested in fixing the documentation issue of this attribute. Do you mean ambiguity instead of ambiguous by chance?

Reply:
> fixing the documentation ...

Improving the docs for this would be appreciated.

> Do you mean ambiguity instead of ambiguous by chance?

Yes, my bad.